src/CodeGenerator/src/Generator/RequestSerializer/RestJsonSerializer.php

```diff
@@ -169,18 +170,27 @@ private function dumpArrayStructure(string $output, string $input, StructureShap
     private function dumpArrayList(string $output, string $input, string $contextProperty, ListShape $shape): string
     {
         $memberShape = $shape->getMember()->getShape();
+        static $counter = -1;
 
-        return strtr('
-            $index = -1;
-            foreach (INPUT as $listValue) {
-                $index++;
-                MEMBER_CODE
-            }
-        ',
-            [
-                'INPUT' => $input,
-                'MEMBER_CODE' => $memberCode = $this->dumpArrayElement(sprintf('%s[$index]', $output), '$listValue', $contextProperty, $memberShape, true),
-            ]);
+        try {
+            ++$counter;
+            $cleanCounter = $counter ?: '';
+
+            return strtr('
+                $indexCOUNTER = -1;
+                foreach (INPUT as $listValueCOUNTER) {
+                    $indexCOUNTER++;
+                    MEMBER_CODE
+                }
+            ',
+                [
+                    'INPUT' => $input,
+                    'COUNTER' => $cleanCounter ?: '',
```

Comment:
Isn't `$cleanCounter` always defined?
```suggestion
                    'COUNTER' => $cleanCounter,
```

Reply:
I see that you want to remove 0. That is why you are doing this.

Reply:
yes. avoid `$index0` (and reduce this PR diff)